solutions/cses-1640.mdx

```diff
@@ -141,3 +123,165 @@ print("IMPOSSIBLE")
 </PySection>
 
 </LanguageSection>
+
+## Solution 2 - Two Pointers
+
+By keeping two pointers, one at each end of the list, we can find the two indices in $\mathcal{O}(N)$ time.
+<LanguageSection>
+<JavaSection>
+
+```java
+import java.io.*;
+import java.util.*;
+
+public class Main {
+	public static void main(String[] args) {
+		FastScanner fs = new FastScanner();
+		long N, X;
+		N = fs.nextInt();
+		X = fs.nextInt();
+
+		List<Pair> a = new ArrayList<>();
+						  
+		// append the element and its index
+		for (int i = 0; i < N; i++) {
+			long x = fs.nextInt();
+			a.add(new Pair(x, i + 1));
+		}
+		Collections.sort(a);
+		int i = 0, j = (int) N - 1;
+		while (i < j) {
+			// adjust the left and right pointers.
+			if (a.get(i).val + a.get(j).val > X)
+				j--;
+			else if (a.get(i).val + a.get(j).val < X)
+				i++;
+			else if (a.get(i).val + a.get(j).val == X) {
+				System.out.println(a.get(i).index + " " + a.get(j).index);
+				return;
+			}
+		}
+		System.out.println("IMPOSSIBLE");
+	}
+
+	static void sort(int[] a) {
+		ArrayList<Integer> l = new ArrayList<>();
+		for (int i : a)
+			l.add(i);
+		Collections.sort(l);
+		for (int i = 0; i < a.length; i++)
+			a[i] = l.get(i);
+	}
+	//BeginCodeSnip{FastScanner}
+	static class FastScanner {
+		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
+		StringTokenizer st = new StringTokenizer("");
+
+		String next() {
+			while (!st.hasMoreTokens())
+			try {
+				st = new StringTokenizer(br.readLine());
+			} catch (IOException e) {
+				e.printStackTrace();
+			}
+			return st.nextToken();
+		}
+
+		int nextInt() {
+			return Integer.parseInt(next());
+		}
+
+		int[] readArray(int n) {
+			int[] a = new int[n];
+			for (int i = 0; i < n; i++)
+			a[i] = nextInt();
+			return a;
+		}
+
+		long nextLong() {
+			return Long.parseLong(next());
+		}
+	}
+	//EndCodeSnip
+
+	static class Pair implements Comparable<Pair> {
+		long val, index;
+
+		public Pair(long x, long y) {
+			this.val = x;
+	  		this.index = y;
+		}
+
+		public int compareTo(Pair o) {
+	  		return -Long.compare(o.val, val);
+		}
+	}
+}
+```
+</JavaSection>
+<CPPSection>
+
+```cpp
+//CodeSnip{CPP Short Template}
+
+int main() {
+	ll N, X;
+	cin >> N >> X;
+	vector<pair<int,int>> a;
+	
+	// append the element and its index
+	for(int i = 0; i < N; i++) {
+		ll x;
+		cin >> x;
+		a.pb({x, i + 1});
+	}
+	
+	sort(a.begin(), a.end());
+	int i = 0, j = N - 1;
+	
+	while(i < j) {
```

Comment:
put a space after the control keyword as well like
```cpp
while (i < j) {

}
```

Reply:
oops ok